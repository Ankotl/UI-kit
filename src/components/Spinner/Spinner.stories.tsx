import React from 'react'

import { getStoryMeta } from '../../utils/storybook/getStoryMeta'
import { getStoryTemplate } from '../../utils/storybook/getStoryTemplate'

import { ISpinnerProps, Spinner as SpinnerCmp } from './Spinner'

const storiesMeta = getStoryMeta({
  component: SpinnerCmp,
})
export default storiesMeta

const storyTemplate = getStoryTemplate((props: ISpinnerProps) => {
  const containerStyle = {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    height: '100vh',
  }

  return (
    <div style={containerStyle}>
      <SpinnerCmp {...props} />
    </div>
  )
})

export const Spinner = storyTemplate({ size: 'md', strokeWidth: 6 })
