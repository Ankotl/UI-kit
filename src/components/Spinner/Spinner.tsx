import React from 'react'

import clsx from 'clsx'

import s from './Spinner.module.scss'

export interface ISpinnerProps {
  className?: string
  size?: 'xs' | 'sm' | 'md' | 'lg' | 'xl'
  strokeWidth?: number
}

export const Spinner: React.FC<ISpinnerProps> = ({
  className,
  strokeWidth = 6,
  size = 'md',
}) => {
  return (
    <svg
      className={clsx(
        s.Spinner,
        {
          [s.Spinner__xl]: size === 'xl',
          [s.Spinner__lg]: size === 'lg',
          [s.Spinner__md]: size === 'md',
          [s.Spinner__sm]: size === 'sm',
          [s.Spinner__xs]: size === 'xs',
        },
        className
      )}
      color="currentColor"
      preserveAspectRatio="xMidYMid"
      viewBox="0 0 100 100"
    >
      <circle
        cx="50"
        cy="50"
        fill="none"
        r="35"
        stroke="currentColor"
        strokeDasharray="164.93361431346415 56.97787143782138"
        strokeWidth={strokeWidth}
      >
        <animateTransform
          attributeName="transform"
          dur="1s"
          keyTimes="0;1"
          repeatCount="indefinite"
          type="rotate"
          values="0 50 50;360 50 50"
        />
      </circle>
    </svg>
  )
}
