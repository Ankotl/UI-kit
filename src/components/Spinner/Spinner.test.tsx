import React from 'react'

import { render } from '@testing-library/react'

import { Spinner } from './Spinner'

describe('Spinner component test', () => {
  test('Has right states', () => {
    const { container } = render(<Spinner />)
    expect(container).toMatchSnapshot()

    const { container: containerSecondary } = render(
      <Spinner size="xl" strokeWidth={8} />
    )
    expect(containerSecondary).toMatchSnapshot()
  })
})
