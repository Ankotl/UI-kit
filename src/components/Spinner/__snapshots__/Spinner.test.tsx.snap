// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner component test Has right states 1`] = `
<div>
  <svg
    class="Spinner Spinner__md"
    color="currentColor"
    preserveAspectRatio="xMidYMid"
    viewBox="0 0 100 100"
  >
    <circle
      cx="50"
      cy="50"
      fill="none"
      r="35"
      stroke="currentColor"
      stroke-dasharray="164.93361431346415 56.97787143782138"
      stroke-width="6"
    >
      <animatetransform
        attributeName="transform"
        dur="1s"
        keyTimes="0;1"
        repeatCount="indefinite"
        type="rotate"
        values="0 50 50;360 50 50"
      />
    </circle>
  </svg>
</div>
`;

exports[`Spinner component test Has right states 2`] = `
<div>
  <svg
    class="Spinner Spinner__xl"
    color="currentColor"
    preserveAspectRatio="xMidYMid"
    viewBox="0 0 100 100"
  >
    <circle
      cx="50"
      cy="50"
      fill="none"
      r="35"
      stroke="currentColor"
      stroke-dasharray="164.93361431346415 56.97787143782138"
      stroke-width="8"
    >
      <animatetransform
        attributeName="transform"
        dur="1s"
        keyTimes="0;1"
        repeatCount="indefinite"
        type="rotate"
        values="0 50 50;360 50 50"
      />
    </circle>
  </svg>
</div>
`;
